edx_exams/apps/api/v1/views.py

```diff
@@ -179,3 +179,78 @@ def patch(self, request, course_id):
             data = {"detail": "Invalid data", "errors": serializer.errors}
 
         return Response(status=response_status, data=data)
+
+
+class CourseExamConfigurationsView(APIView):
+    # todo: make parameters descriptions more useful.
+    """
+    View to create and update course exam configs for a specific course.
+
+    Given a course id and a proctoring provider, this view will either create a new course exam configuration
+    (if one doesn't exist), or modify the proctoring provider on an existing course exam config.
+
+    HTTP PATCH
+    Creates or updates a CourseExamConfiguration.
+    Expected PATCH data: {
```

Comment:
You're right @MichaelRoytman! I was a bit confused on that but thanks for helping clarify! So I've updated the params to only include provider name, so I'm guessing PATCH is still right?

Reply:
Is patch still the right request type here?

Reply:
I think a PUT technically makes sense because it 'replaces' the entire object. If we add additional values (such as the escalation email) I'd expect that the UI to update all of them at once.

Reply:
I think a PUT requires that you supply a complete representation of the object to be replaced. If you're supplying just a single field (i.e. `name`), that would make it a PATCH.

Reply:
So the way I'm looking at this CourseExamConfiguration is the resource being replaced, ProctoringProvider is just a related object.

Reply:
Based on [this document](https://openedx.atlassian.net/wiki/spaces/AC/pages/18350757/edX+REST+API+Conventions), `PATCH` seems a better fit to me as well. Does that sound good, @Zacharis278? Definitely open to thoughts on the advantages to using `PUT` here. Thanks Zach and @MichaelRoytman for the thoughts here!

Reply:
yeah, I suppose PATCH is a better catch-all anyway

Reply:
Sounds good!